tests/unbound_type_variables.rs

```diff
@@ -57,4 +57,8 @@ fn unbound_type_variables() {
                        f null",
         "a"
     );
+    assert_unbound!(
+        "let rec f = fun x => if x == 0 then x else f (x - n) in f null",
+        "n"
+    );
```

Comment:
This test is succeeding a bit by accident, I think. This file is testing for unbound *type variables*, that is unbound variables that happens in a type position (right hand side of a type or contract annotation). Here there is no annotation whatsoever, but because some unbound type variables manifest as normal unbound variables, the macro `assert_unbound!` is liberal enough to make it work.
Beside this test being useful or not, it shouldn't be here. Additionally, I'm not sure it is testing something new that isn't already tested (there is no reason recursive lets would make unbound variables to be suddenly bound). I would suggest to just get rid of it.

To see if the `free_var` transform is doing its job correctly on recursive lets, you can add new test cases in  `tests/free_var.rs`.